### DISCUSSION TITLE
Feature Request: Output when an imported resource is also changed in the same plan

### DISCUSSION THREAD

**Discussion by DylanRodgers98** [suzuki-shunsuke/tfcmt#930] [Deprecated (Feature Request)] [answered]
### Feature Overview

Currently imports are only output if there are no subsequent changes to them. If an import is changed within the same plan they are output as updates.

For example, with a plan like:
```tf
  # resource.default will be updated in-place
  # (imported from "my_resource")
  ~ resource "resource" "default" {
        id = "my_resource"
      - labels = {}
    }

Plan: 1 to import, 0 to add, 1 to change, 0 to destroy.
```

The following is output by tfcmt:
```
* Update
  * resource.default
```

This feature would include the resource in the list of imports:
```
* Import
  * resource.default
* Update
  * resource.default
```

### Why is the feature needed?

It would be very useful to see if an imported resource is changed within the same plan.

### Does the feature include Breaking Changes?

Not as far as I'm aware.

**Comment by suzuki-shunsuke** [comment]
Thank you for your feedback!

I'll take a look.

https://github.com/suzuki-shunsuke/tfcmt/blob/912cb43097cadeb8f0048b3dc614b474f3206326/pkg/terraform/parser.go#L73

https://github.com/suzuki-shunsuke/tfcmt/blob/912cb43097cadeb8f0048b3dc614b474f3206326/pkg/terraform/parser.go#L176-L177

**Reply by suzuki-shunsuke** [reply]
moved block has also the same issue.

```tf
  # github_repository.tfaction-2 will be updated in-place
  # (moved from github_repository.tfaction)
```

**Reply by suzuki-shunsuke** [reply]
`replace` should also be supported.

```tf
  # github_issue.test-2 must be replaced
  # (moved from github_issue.test)
```

**Comment by suzuki-shunsuke** [comment]
There are several options.

1. Add Import

```
* Import
  * resource.default
* Update
  * resource.default
```

2. Add `(imported from ~)` to Update

```
* Update
  * resource.default (imported from "my_resource")
```

3. Both

```
* Import
  * resource.default
* Update
  * resource.default (imported from "my_resource")
```

**Reply by suzuki-shunsuke** [reply]
I'll adopt 1 for now.

- The implementation is simple
- For 2 and 3, we need to take into account the compatibility of template variables

**Comment by suzuki-shunsuke** [comment]
- https://github.com/suzuki-shunsuke/tfcmt/pull/932

**Reply by DylanRodgers98** [reply]
Nice, thank you for implementing this!

**Reply by suzuki-shunsuke** [reply, marked as answer]
tfcmt v4.6.0 is out 🚀
https://github.com/suzuki-shunsuke/tfcmt/releases/tag/v4.6.0
